api/db/changelog/db.changelog-44-notebook_cohort_cache.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+    <changeSet author="nehasaxena" id="db.changelog-44-cacheTable">
+        <createTable tableName="notebook_cohort_cache">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="user_workspace_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+            <column name="cohort_id" type="bigint">
+            </column>
+            <column name="notebook_name" type="varchar(255)">
+            </column>
```

Comment:
Having one column per resource type seems inefficient. Could we have one column for "resource_identifier" and then let the calling code (service class) determine what identifier actually represents?

If you did take this approach, you'd also need to add a string column for resource type that could come from an Enum class. You'd need this in the case that two different resources happened to be identified by the same string identifier, and you'd need to disambiguate that.

Reply:
I did think about this earlier, but by having cohort as fk I am gaining the functionality of safe delete as soon as Cohorts are deleted.

Reply:
Great point. Feel free to ignore this suggestion then.